src/rtc_monitor.ts

```diff
@@ -0,0 +1,261 @@
+                    candidate = newRTCCandidatePairStats(report);
+                }
+            }
+
+            if (report.type === 'inbound-rtp' && report.kind === 'audio') {
+                localIn[report.ssrc] = newRTCLocalInboundStats(report);
+            }
+
+            if (report.type === 'remote-inbound-rtp' && report.kind === 'audio') {
+                remoteIn[report.ssrc] = newRTCRemoteInboundStats(report);
+            }
+        });
+
+        if (!candidate) {
+            this.logger.logDebug('RTCMonitor: no valid candidate was found');
+            return;
+        }
+
+        // Step 1: get transport latency from the in-use candidate pair stats, if present.
+        let transportLatency;
+
+        // currentRoundTripTime could be missing in the original report (e.g. on Firefox) and implicitly coverted to NaN.
+        if (!isNaN(candidate.currentRoundTripTime)) {
+            transportLatency = (candidate.currentRoundTripTime * 1000) / 2;
+        }
+
+        // Step 2: if receiving any stream, calculate average jitter and loss rate using local stats.
+        const localInStats = this.getLocalInQualityStats(localIn);
+
+        // Step 3: if sending any stream, calculate average latency, jitter and
+        // loss rate using remote stats.
+        const remoteInStats = this.getRemoteInQualityStats(remoteIn);
+
+        // Step 4: cache current stats for calculating deltas on next iteration.
+        this.stats.lastLocalIn = {
+            ...localIn,
+        };
+        this.stats.lastRemoteIn = {
+            ...remoteIn,
+        };
+
+        if (typeof transportLatency === 'undefined' && typeof remoteInStats.avgLatency === 'undefined') {
+            transportLatency = this.peer.getRTT() / 2;
+        }
+
+        if (typeof localInStats.avgJitter === 'undefined' && typeof remoteInStats.avgJitter === 'undefined') {
+            this.logger.logDebug('RTCMonitor: jitter could not be calculated');
+            return;
+        }
+
+        if (typeof localInStats.avgLossRate === 'undefined' && typeof remoteInStats.avgLossRate === 'undefined') {
+            this.logger.logDebug('RTCMonitor: lossrate could not be calculated');
+            return;
+        }
+
+        const jitter = Math.max(localInStats.avgJitter || 0, remoteInStats.avgJitter || 0);
+        const lossRate = Math.max(localInStats.avgLossRate || 0, remoteInStats.avgLossRate || 0);
+        const latency = transportLatency ?? remoteInStats.avgLatency;
+
+        // Step 5 (or the magic step): calculate MOS (Mean Opinion Score)
+        const mos = this.calculateMOS(latency!, jitter, lossRate);
+        this.emit('mos', mos);
+        this.logger.logDebug(`RTCMonitor: MOS --> ${mos}`);
+    }
+
+    private calculateMOS(latency: number, jitter: number, lossRate: number) {
+        this.logger.logDebug(`RTCMonitor: MOS inputs --> latency: ${latency} jitter: ${jitter} loss: ${lossRate}`);
```

Comment:
Keeping this logs for now but we may consider suppressing them once we tested the changes since they will fire every couple of seconds.